src/libs/ReportUtils.js

```diff
@@ -762,10 +781,11 @@ function getIOUReportActionMessage(type, total, participants, comment, currency)
  * @param {String} paymentType - Only required if the IOUReportAction type is 'pay'. Can be oneOf(elsewhere, payPal, Expensify).
  * @param {String} iouTransactionID - Only required if the IOUReportAction type is oneOf(cancel, decline). Generates a randomID as default.
  * @param {String} iouReportID - Only required if the IOUReportActions type is oneOf(decline, cancel, pay). Generates a randomID as default.
+ * @param {String} isSettlingUp - Whether we are settling up an IOU.
  *
  * @returns {Object}
  */
-function buildOptimisticIOUReportAction(sequenceNumber, type, amount, currency, comment, participants, paymentType = '', iouTransactionID = '', iouReportID = '') {
+function buildOptimisticIOUReportAction(sequenceNumber, type, amount, currency, comment, participants, paymentType = '', iouTransactionID = '', iouReportID = '', isSettlingUp) {
```

Comment:
I'm not sure about this dangling argument here, open to suggestions on how to make this cleaner. `buildOptimisticIOUReportAction` needs to be aware of whether we are sending money or paying an IOU in order to display the proper message.

Reply:
1. The parameter should have a default value for sure, otherwise we should move it infront of the optional params.
2. I think we have a "bug" in [here](https://github.com/Expensify/App/pull/12739/files#diff-65c096044d5f69b35bcdec14c99c4fda4580759df9b1a7c36650d58eea276f1dR805) since if you pay money request we do not have the IOUDetails field in backend 
<img width="879" alt="image" src="https://user-images.githubusercontent.com/36083550/205622056-a0dde853-3faf-4e6e-80be-4760b0dbadff.png">
This si a screenshot of how the data looks like coming from the server.

So we should also update that since this field should be added only if we are Sending money

I think in a hindsight it might be easier to have a specific payment type for sending money and settling up.

However, I think what we can do here and would be easier is to use the `iouTransactionID` and or `iouReportID` parameters. Those should be truthy only if we are paying money request/ settling up. If they are not passed in AND the type is PAY, then we are sending money. If you include some comment I think that will be easier and nicer than adding another parameter. 

Does it make sense? @youssef-lr ?

Reply:
> However, I think what we can do here and would be easier is to use the iouTransactionID and or iouReportID parameters. 
I'm not sure this would work @vitHoracek , we also send the `iouReportID` when sending money, and `iouTransactionID` is only used when we cancel a money request. I worked on a solution locally which is to the `iouReport` from Onyx and if it `hasOutstandingIOU` then it means we're settling up. What do you think?

Reply:
OK I tried your suggestion but then the IOUPreview keeps loading, I guess the component is not getting the `iouReportID`.

<img width="512" alt="image" src="https://user-images.githubusercontent.com/9680864/206205610-a056d030-2665-496f-bf71-be893aa13315.png">